### PR TITLE
Add anomaly detection

### PR DESCRIPTION
**Pull request by maniax89**

### CODE REVIEW

File: src/AnomalyDetection/index.jsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    }).isRequired,
+    colorAnomaly: string.isRequired,
+    colorAnomalyActive: string.isRequired,
+    colorLine: string.isRequired,
+  }
+
+  static defaultProps = {
+    colorAnomaly: '#8c101c',
+    colorAnomalyActive: '#ff5050',
+    colorLine: '#00a78f',
```
Comment (besh) [inline src/AnomalyDetection/index.jsx:37]: Interesting way to handle styling.

File: src/AnomalyDetection/index.jsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+  onShowResults = () => {
+    this.setState({ showQuery: false });
+  }
+
+  renderDot = (dotProps, color) => {
+    const dotColor = color || this.props.colorAnomaly;
+    const { payload } = dotProps;
+    const newDotProps = Object.assign({}, dotProps, {
+      fill: AnomalyDetection.hasAnomaly(payload) ? dotColor : dotProps.fill,
```
Comment (besh) [inline src/AnomalyDetection/index.jsx]: maybe 

```js
const hasAnomaly = AnomalyDetection.hasAnomaly(payload);
const newDotProps = Object.assign({}, dotProps, {
  fill: hasAnomaly ? dotColor: dotProps.fill,
  fill: hasAnomaly ? dotColor: dotProps.stroke
});
```

File: src/__test__/AnomalyDetection/index.test.jsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+    describe('when tooltipProps has an anomaly', () => {
+      const tooltipProps = {
+        payload: [
+          {
+            payload: {
+              anomaly: 0.5,
+            },
+          },
+        ],
```
Comment (besh) [inline src/__test__/AnomalyDetection/index.test.jsx:92]: All of this comma-dangle. Is this from the airbnb linter?

Comment (maniax89) [reply src/__test__/AnomalyDetection/index.test.jsx:92]: yes

**Pull request merged**